src/metabase/query_processor.clj

```diff
@@ -74,58 +74,63 @@
                          [column-level-perms-check :as ee.sandbox.columns]
                          [row-level-restrictions :as ee.sandbox.rows]]))
 
-;; ▼▼▼ POST-PROCESSING ▼▼▼  happens from TOP-TO-BOTTOM, e.g. the results of `f` are (eventually) passed to `limit`
+;; ▼▼▼ POST-PROCESSING ▼▼▼  happens from TOP-TO-BOTTOM
 (def default-middleware
   "The default set of middleware applied to queries ran via `process-query`."
-  [#'mbql-to-native/mbql->native
-   #'check-features/check-features
-   #'limit/limit
-   #'cache/maybe-return-cached-results
-   #'optimize-temporal-filters/optimize-temporal-filters
-   #'validate-temporal-bucketing/validate-temporal-bucketing
-   #'auto-parse-filter-values/auto-parse-filter-values
-   #'wrap-value-literals/wrap-value-literals
-   #'annotate/add-column-info
-   #'perms/check-query-permissions
-   #'pre-alias-ags/pre-alias-aggregations
-   #'cumulative-ags/handle-cumulative-aggregations
+  [#'mbql-to-native/mbql->native                                      ; EXECUTE
```

Comment:
I guess too late but I think I would have done something like

```clojure
(m.43/wrap #'limit/limit-result-rows-middleware #'limit/add-default-limit-middleware)
(m.43/wrap identity #'annotate/add-column-info)
```

then when we undo this the middleware's don't change, and we're sure we are using the same vars in the new middleware stacks that will emerge. And the comments cannot drift or be incorrect.

Reply:
yeah that actually does look like a better idea in hindsight